web/index.js

```diff
@@ -77,35 +107,130 @@ map.on('load', function() {
 
 
 function filter_on_one_route(route) {
-    var ways_ids = route['ways_ids'];
-    var positions_ids = route['positions_ids'];
-    map.setFilter('routes_ways_filtered_outline', ["in", "id"].concat(ways_ids || []));
-    map.setFilter('routes_ways_filtered', ["in", "id"].concat(ways_ids || []));
-    map.setFilter('routes_points_filtered', ["in", "id"].concat(positions_ids || []));
-    map.setPaintProperty('routes_ways_filtered', "line-color", route['rel_colour'] || 'grey')
-    map.setPaintProperty('routes_points_filtered', "circle-color", route['rel_colour'] || 'grey')
-    html = `<div class='bus_box_div'>
-                <span class='bus_box' style='border-bottom-color: ${route['rel_colour'] || "grey"};' >
+    reset_filters_and_show_all_lines();
+
+    var caisson_content = `
+    <div id="route_info"></div>
+    <div id="close_caisson_button"></div>
+    <div id="stop_list" class="stop_list"></div>
+    <div id="osm_attribution"></div>
+    `
+    caisson.add_content(caisson_content)
+    var close_caisson_button = document.getElementById('close_caisson_button')
+    close_caisson_button.innerHTML = `<a href='#' onclick='reset_filters_and_show_all_lines()'>Masquer la ligne</a>`;
+
+    const route_id = route['rel_osm_id'];
+
+    const stop_list_url = vapour_trail_api_base_url + "/routes/" + route_id;
+    fetch(stop_list_url)
+        .then(function(data) {
+            return data.json()
+        })
+        .then(function(route_data) {
+            var route_colour = route_data['route_info']['colour'] || 'grey';
+            var route_info = document.getElementById('route_info');
+            route_info.innerHTML = create_route_medata(route_data['route_info']);
+
+            var thermo = create_stop_list_for_a_route(route_data['stop_list'], route_colour)
+            var stop_list = document.getElementById('stop_list')
+            stop_list.innerHTML = thermo;
+
+            var osm_attribution = document.getElementById('osm_attribution')
+            osm_attribution.innerHTML = create_osm_attribution_for_the_route(route_data['route_info']['osm_id']);
```

Comment:
The `osm_id` here actually is the imposm id (`-6434463` instead of `6434463`). Do you think we should fix it in the API or in the frontend ?

Reply:
I would prefer showing consistent `osm_id` and not relying on internal imposm id if possible.
(if someone someday wants to use both Vapour api and Overpass for exemple).